### PR TITLE
[rps] Resintate tests

### LINKED ISSUE #713: [rps] AutoOpponent/Rps Channel Client tests need to be fixed
Currently the tests in `rps-channel-client.test.ts` and `auto-opponent.test.ts` are disabled. They were failing after the switch in `rps` to use a real `channelClient`. They probably need to just mock the `channel-client` before running.

### PR DESCRIPTION
**Pull request by geoknee**
Closes #713 .

This PR undisables these tests, which were set to "skip" recently. 

1. The auto-opponent and test use the `FakeChannelClient`

2. The test for `RPSChannelClient` class mocks up a `ChannelClient` and does not assert anything that requires the underlying provider to exist (for example, an event being emitted). It also expands the test coverage to 

 - [x] `onMessageQueued()`
 - [x] `onChannelUpdated()`
 - [x] `createChannel()`
 - [x] `joinChannel()`
 - [x] `closeChannel()`
 - [x] `updateChannel()`
 - [x] `pushMessage()`
 - [x] `getAddress()`

👀 Additional modifications to non-test code:
- A wrapper `Client` will not spawn the underlying `Client`: it must be passed in, instead (makes testing via mocks far easier)
- (related) Channel Client's `enable` method no longer does anything.
- Encoding mismatch fixed (uncovered by new testing approach).
- Wrapper client's `pushMessage` method passes through (i.e. returns) the return value of the inner class instance's `pushMessage`

### CODE REVIEW

File: packages/rps/src/redux/store.ts [hunk 1]
```diff
@@ -26,9 +27,7 @@ function* rootSaga() {
   const channelProviderEnabled = yield stateChannelWalletSaga();
 
   if (channelProviderEnabled) {
-    const client = new RPSChannelClient();
-
-    yield call([client, client.enable]);
+    const client = new RPSChannelClient(new ChannelClient(window.channelProvider));
```
Comment (geoknee) [inline packages/rps/src/redux/store.ts]: These lines encapsulate fairly well the new approach (of this PR) to instantiating the double wrapper around `window.channelProvider`. There were comments in the existing code intimating that this was the desired approach in any case, and it also makes testing far easier.

File: packages/rps/src/utils/rps-channel-client.ts [hunk 2]
```diff
@@ -1,21 +1,15 @@
...
 import {RPS_ADDRESS} from '../constants';
 import {bigNumberify} from 'ethers/utils';
 
 // This class wraps the channel client converting the request/response formats to those used in the app
 
 export class RPSChannelClient {
-  channelClient: ChannelClientInterface;
+  constructor(private readonly channelClient: ChannelClientInterface) {}
 
   async enable() {
```
Comment (cf19drofxots) [inline packages/rps/src/utils/rps-channel-client.ts]: Why not remove this method?

Comment (geoknee) [reply packages/rps/src/utils/rps-channel-client.ts]: It's not obvious from the syntax, but this constructor is going to set `this.channelClient` to what you pass in to it. So it is the same as writing {`this.channelClient = channelClient`} in the body. https://www.typescriptlang.org/docs/handbook/classes.html#parameter-properties

Comment (geoknee) [reply packages/rps/src/utils/rps-channel-client.ts]: (and we need `this.channelClient` to be available to most of the class methods).

Comment (cf19drofxots) [reply packages/rps/src/utils/rps-channel-client.ts]: Oh hah no I meant line 11 with the function implementation of `enable` being empty. I get what's happening on the line you explained :)

**Pull request merged**